src/fluree/db/query/fql/syntax.cljc

```diff
@@ -60,146 +59,161 @@
-     ::filter               [:sequential ::function]
-     ::union                [:sequential [:sequential ::where-pattern]]
-     ::bind                 [:map-of ::var :any]
-     ::where                [:sequential [:orn
-                                          [:where-map ::where-map]
-                                          [:tuple ::where-tuple]]]
-     ::var-collection       [:sequential ::var]
-     ::val-collection       [:sequential ::val]
-     ::single-var-binding   [:tuple ::var ::val-collection]
-     ::value-binding        [:sequential ::val]
-     ::multiple-var-binding [:tuple
-                             ::var-collection
-                             [:sequential ::value-binding]]
-     ::values               [:orn
-                             [:single ::single-var-binding]
-                             [:multiple ::multiple-var-binding]]
-     ::t                    [:or :int :string]
-     ::context              ::v/context
-     ::analytical-query     [:map {:decode/json decode-json}
+                                       [:val ::subject]]]
+                            [:predicate [:orn
+                                         [:var ::var]
+                                         [:iri ::iri]]]
+                            [:object [:orn
+                                      [:var ::var]
+                                      [:ident [:fn pred-ident?]]
+                                      [:iri-map ::iri-map]
+                                      [:val :any]]]]
+    ::where-tuple          [:orn
+                            [:triple ::triple]
+                            [:remote [:sequential {:max 4} :any]]]
+    ::where-pattern        [:orn
+                            [:map ::where-map]
+                            [:tuple ::where-tuple]]
+    ::optional             [:orn
+                            [:single ::where-pattern]
+                            [:collection [:sequential ::where-pattern]]]
+    ::filter               [:sequential ::function]
+    ::union                [:sequential [:sequential ::where-pattern]]
+    ::bind                 [:map-of ::var :any]
+    ::where                [:sequential [:orn
+                                         [:where-map ::where-map]
+                                         [:tuple ::where-tuple]]]
+    ::var-collection       [:sequential ::var]
+    ::val-collection       [:sequential ::val]
+    ::single-var-binding   [:tuple ::var ::val-collection]
+    ::value-binding        [:sequential ::val]
+    ::multiple-var-binding [:tuple
+                            ::var-collection
+                            [:sequential ::value-binding]]
+    ::values               [:orn
+                            [:single ::single-var-binding]
+                            [:multiple ::multiple-var-binding]]
+    ::t                    [:or :int :string]
+    ::context              ::v/context
+    ::json-ld-keyword      [:keyword {:decode/json decode-json-ld-keyword
```

Comment:
I think we'll need this `::json-ld-keyword` in the history query schema as well.